enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -132,21 +118,31 @@ def _setup_enclave(self):
         functionalities over this instance of the enclave. Therefore
         all relevant data is sent back here. This is kept in memory
         for use hereafter.
-
-        Returns :
-            signup_data - Relevant signup data to be used for requests to the
-                          enclave
         """
         try:
             logger.info("Initialize enclave and create signup data")
             signup_data = self._create_signup_data()
             if signup_data is None:
                 logger.error("Failed to create signup data")
-                return None
+                raise Exception("Failed to create signup data")
         except Exception as e:
-            logger.exception("failed to initialize/signup enclave; %s", str(e))
+            logger.exception("Failed to initialize/signup enclave; %s", str(e))
             sys.exit(-1)
-        return self._get_JSON_from_signup_object(signup_data)
+
+        self.enclave_data = self._get_JSON_from_signup_object(signup_data)
+        self.sealed_data = self.enclave_data["sealed_data"]
+        self.verifying_key = self.enclave_data["verifying_key"]
+        self.encryption_key = self.enclave_data["encryption_key"]
+        self.encryption_key_signature = \
+            self.enclave_data["encryption_key_signature"]
+        self.enclave_id = self.enclave_data["enclave_id"]
+        self.proof_data = self.enclave_data["proof_data"]
+        self.extended_measurements = self.enclave_data["measurements"]
+
+        # TODO: encryption_key_nonce is hardcoded to an empty str
+        # Need to come up with a scheme to generate it for every unique
+        # encryption key.
+        self.encryption_key_nonce = ""
```

Comment:
Why not just generate a 24 byte/192 bit random number?

Reply:
Done